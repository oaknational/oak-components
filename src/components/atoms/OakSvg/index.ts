export { OakSvg } from "./OakSvg";
export type { OakSvgProps, OakSvgNames } from "./OakSvg";
