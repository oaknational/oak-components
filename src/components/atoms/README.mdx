# Atoms

## Rules for components in this folder

- Can import other atoms
- Imports from molecules and organisms are forbidden!
- Generic
- Unstyled
- Avoid embedded logic and state
- No sub-components
  (eg. Box)
