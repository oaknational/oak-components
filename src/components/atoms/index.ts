export * from "./OakBox";
export * from "./OakFlex";
export * from "./OakSpan";
export * from "./OakForm";
export * from "./OakImage";
export * from "./OakTypography";
export * from "./OakHeading";
export * from "./OakP";
export * from "./OakUL";
export * from "./OakOL";
export * from "./OakLI";
export * from "./OakLabel";
export * from "./OakIcon";
export * from "./OakGrid";
export * from "./OakGridArea";
export * from "./OakThemeProvider";
export * from "./OakMaxWidth";
export * from "./OakCloudinaryImage";
export * from "./OakKbd";
export * from "./OakGlobalStyle";
export * from "./OakAnchorTarget";
export * from "./OakFieldSet";
