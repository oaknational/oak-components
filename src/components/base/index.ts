export * from "./OakBox";
export * from "./OakFlex";
export * from "./OakBaseCard";
export * from "./OakSpan";
export * from "./OakForm";
export * from "./OakImage";
export * from "./OakTypography";
export * from "./OakHeading";
export * from "./OakP";
export * from "./OakUL";
export * from "./OakOL";
export * from "./OakLI";
export * from "./OakLabel";
