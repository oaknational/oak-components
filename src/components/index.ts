export * from "./base";
// export * from "./ui"; // uncomment this line once there are components in this folder
// export * from "./integrated"; // uncomment this line once there are components in this folder
