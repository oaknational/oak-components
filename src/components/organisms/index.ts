export * from "./pupil";
export * from "./teacher";
export * from "./OakSubjectIcon";
export * from "./OakPrimaryNavItem";
export * from "./OakPrimaryNav";
export * from "./OakCookieBanner";
export * from "./OakCookieSettingsModal";
export * from "./OakCookieConsentProvider";
export * from "./OakCookieConsent";
