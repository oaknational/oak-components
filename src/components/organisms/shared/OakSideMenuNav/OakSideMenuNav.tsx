import React, { useState } from "react";
import styled from "styled-components";

import {
  OakFlex,
  OakHeading,
  OakIcon,
  OakLI,
  OakSpan,
  OakUL,
} from "@/components/atoms";
import { flexStyle, FlexStyleProps } from "@/styles/utils/flexStyle";
import { parseDropShadow } from "@/styles/helpers/parseDropShadow";
import { paddingStyle, PaddingStyleProps } from "@/styles/utils/spacingStyle";
import { parseColor } from "@/styles/helpers/parseColor";
import { getBreakpoint } from "@/styles/utils/responsiveStyle";

const StyledNav = styled.nav`
  outline: none;
`;

const StyledLink = styled("a")<
  FlexStyleProps & PaddingStyleProps & { isSelected: boolean }
>`
  text-decoration: none;
  display: flex;
  outline: none;

  @media (min-width: ${getBreakpoint("small")}px) {
    border-left: ${(props) =>
      props.isSelected ? "4px solid #222222" : "4px solid transparent"};
    :hover {
      text-decoration: underline;
      border-color: ${(props) =>
        props.isSelected
          ? parseColor("bg-btn-primary-hover")
          : parseColor("bg-decorative1-main")};
    }
  }

  :focus-visible {
    box-shadow: ${parseDropShadow("drop-shadow-centered-lemon")},
      ${parseDropShadow("drop-shadow-centered-grey")};
    border-color: transparent;
    border-radius: 4px;
  }

  ${flexStyle}
  ${paddingStyle}
`;

type MenuItemProps = {
  heading: string;
  subheading?: string;
  href: string;
};

export type OakSideMenuNavProps = {
  heading: string;
  menuItems: MenuItemProps[];
};

export const OakSideMenuNav = (props: OakSideMenuNavProps) => {
  const { heading, menuItems } = props;
  const [selectedHref, setSelectedHref] = useState<string | null>(null);

  return (
    <StyledNav aria-labelledby="side-menu-header">
      <OakFlex
        $flexDirection="column"
        $background={["bg-decorative1-subdued", "white"]}
        $pa={["inner-padding-xl4", "inner-padding-none"]}
        $gap="space-between-s"
      >
        <OakHeading
          tag="h2"
          $font="heading-light-7"
          $color="text-subdued"
          id="side-menu-header"
        >
          {heading}
        </OakHeading>
        <OakUL
          $reset
          $display="flex"
          $gap="space-between-s"
          $flexDirection="column"
        >
          {menuItems.map((item) => (
            <OakLI key={item.heading}>
              <StyledLink
                $alignItems={["center", "flex-start"]}
                $columnGap="space-between-s"
                href={item.href}
                $ph={["inner-padding-none", "inner-padding-s"]}
                isSelected={selectedHref === item.href}
                $flexDirection={["row", "column"]}
                onClick={() => setSelectedHref(item.href)}
                aria-current={selectedHref === item.href ? "true" : undefined}
              >
                <OakFlex
                  $flexDirection={["row", "column"]}
                  $columnGap={["space-between-s", "space-between-sssx"]}
                  $flexWrap="wrap"
                >
                  <OakSpan $font="heading-light-7" $color="text-primary">
                    {item.heading}
                  </OakSpan>
                  {item.subheading && (
                    <OakSpan $font="body-3" $color="text-subdued">
                      {item.subheading}
                    </OakSpan>
                  )}
                </OakFlex>
                <OakIcon
                  $display={["block", "none"]}
                  iconName="chevron-right"
                  $width="all-spacing-6"
                  $pl="inner-padding-s"
                />
              </StyledLink>
            </OakLI>
          ))}
        </OakUL>
      </OakFlex>
    </StyledNav>
  );
};
