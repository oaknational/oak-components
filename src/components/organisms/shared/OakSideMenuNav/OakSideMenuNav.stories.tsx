import React from "react";
import { StoryObj, Meta } from "@storybook/react";

import OakSideMenuNav from "./OakSideMenuNav";

import { OakGrid, OakGridArea } from "@/components/atoms";

const meta: Meta<typeof OakSideMenuNav> = {
  component: OakSideMenuNav,
  tags: ["autodocs"],
  argTypes: {
    heading: {
      control: {
        type: "text",
      },
    },
  },
};

export default meta;

type Story = StoryObj<typeof OakSideMenuNav>;

export const Default: Story = {
  render: (args) => (
    <OakGrid $background="pink" $width="100%" $height="100vh">
      <OakGridArea $colSpan={[12, 2]}>
        <OakSideMenuNav {...args} />
      </OakGridArea>
    </OakGrid>
  ),
  args: {
    heading: "Test Heading",
    menuItems: [
      {
        heading: "Test Item 1",
        subheading: "Test Subheading 1",
        href: "#test1",
      },
      {
        heading: "Test Item 2",
        subheading: "Test Subheading 2",
        href: "#test2",
      },
      {
        heading: "Test Item 3",
        subheading: "Test Subheading 3",
        href: "#test3",
      },
      {
        heading: "Test Item 4",
        subheading: "Test Subheading 4",
        href: "#test4",
      },
    ],
  },
};
