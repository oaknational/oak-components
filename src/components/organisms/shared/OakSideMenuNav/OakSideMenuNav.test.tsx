import React from "react";
import "@testing-library/jest-dom";
import { screen } from "@testing-library/react";
import { create } from "react-test-renderer";

import OakSideMenuNav from "./OakSideMenuNav";

import renderWithTheme from "@/test-helpers/renderWithTheme";
import { OakThemeProvider } from "@/components/atoms";
import { oakDefaultTheme } from "@/styles";

const baseProps = {
  heading: "Test Heading",
  menuItems: [
    {
      heading: "Test Item 1",
      subheading: "Test Subheading 1",
      href: "#test1",
    },
    {
      heading: "Test Item 2",
      subheading: "Test Subheading 2",
      href: "#test2",
    },
  ],
};

describe("OakSideMenuNav", () => {
  it("renders a heading", () => {
    renderWithTheme(<OakSideMenuNav {...baseProps} />);
    const heading = screen.getByRole("heading", { name: /Test Heading/i });
    expect(heading).toBeInTheDocument();
  });
  it("matches snapshot", () => {
    const tree = create(
      <OakThemeProvider theme={oakDefaultTheme}>
        <OakSideMenuNav {...baseProps} />,
      </OakThemeProvider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
