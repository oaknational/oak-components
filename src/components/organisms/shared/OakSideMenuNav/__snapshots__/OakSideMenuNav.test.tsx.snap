// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OakSideMenuNav matches snapshot 1`] = `
[
  .c1 {
  padding: 3rem;
  background: #dff9de;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c8 {
  font-family: --var(google-font),Lexend,sans-serif;
}

.c12 {
  position: relative;
  width: 1.5rem;
  min-width: 1.5rem;
  height: 2rem;
  min-height: 2rem;
  padding-left: 0.75rem;
  display: block;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 1rem;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-column-gap: 1rem;
  column-gap: 1rem;
}

.c13 {
  object-fit: contain;
}

.c3 {
  font-family: --var(google-font),Lexend,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.25rem;
  -webkit-letter-spacing: 0.0115rem;
  -moz-letter-spacing: 0.0115rem;
  -ms-letter-spacing: 0.0115rem;
  letter-spacing: 0.0115rem;
  color: #575757;
}

.c10 {
  color: #222222;
  font-family: --var(google-font),Lexend,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.25rem;
  -webkit-letter-spacing: 0.0115rem;
  -moz-letter-spacing: 0.0115rem;
  -ms-letter-spacing: 0.0115rem;
  letter-spacing: 0.0115rem;
}

.c11 {
  color: #575757;
  font-family: --var(google-font),Lexend,sans-serif;
  font-weight: 300;
  font-size: 0.875rem;
  line-height: 1.25rem;
  -webkit-letter-spacing: -0.005rem;
  -moz-letter-spacing: -0.005rem;
  -ms-letter-spacing: -0.005rem;
  letter-spacing: -0.005rem;
  white-space: break-spaces;
}

.c5 {
  display: revert;
  font-family: --var(google-font),Lexend,sans-serif;
  font-family: --var(google-font),Lexend,sans-serif;
  display: revert;
  display: revert;
}

.c4 {
  list-style: none;
  padding: 0;
  margin: 0;
  display: block;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-family: --var(google-font),Lexend,sans-serif;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 1rem;
}

.c6 {
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  outline: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-column-gap: 1rem;
  column-gap: 1rem;
  padding-left: 0rem;
  padding-right: 0rem;
}

.c6:focus-visible {
  box-shadow: 0 0 0 0.125rem rgba(255,229,85,100%), 0 0 0 0.3rem rgba(87,87,87,100%);
  border-color: transparent;
  border-radius: 4px;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 {
  outline: none;
}

@media (min-width:750px) {
  .c1 {
    padding: 0rem;
  }
}

@media (min-width:750px) {
  .c1 {
    background: #ffffff;
  }
}

@media (min-width:750px) {
  .c12 {
    display: none;
  }
}

@media (min-width:750px) {
  .c9 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (min-width:750px) {
  .c9 {
    -webkit-column-gap: 0.25rem;
    column-gap: 0.25rem;
  }
}

@media (min-width:750px) {
  .c6 {
    border-left: 4px solid transparent;
  }

  .c6:hover {
    -webkit-text-decoration: underline;
    text-decoration: underline;
    border-color: #bef2bd;
  }
}

@media (min-width:750px) {
  .c6 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (min-width:750px) {
  .c6 {
    -webkit-align-items: flex-start;
    -webkit-box-align: flex-start;
    -ms-flex-align: flex-start;
    align-items: flex-start;
  }
}

@media (min-width:750px) {
  .c6 {
    padding-left: 0.75rem;
  }
}

@media (min-width:750px) {
  .c6 {
    padding-right: 0.75rem;
  }
}

<nav
    aria-labelledby="side-menu-header"
    className="c0"
  >
    <div
      className="c1 c2"
    >
      <h2
        className="c3"
        id="side-menu-header"
      >
        Test Heading
      </h2>
      <ul
        className="c4"
      >
        <li
          className="c5"
        >
          <a
            className="c6 c7"
            href="#test1"
            onClick={[Function]}
          >
            <div
              className="c8 c9"
            >
              <span
                className="c10"
              >
                Test Item 1
              </span>
              <span
                className="c11"
              >
                Test Subheading 1
              </span>
            </div>
            <div
              className="c12"
            >
              <img
                alt=""
                className="c13"
                data-nimg="fill"
                decoding="async"
                loading="lazy"
                onError={[Function]}
                onLoad={[Function]}
                src="https://res.cloudinary.com/mock-cloudinary-cloud/image/upload/v1707752509/icons/vk9xxxhnsltsickom6q9.svg"
                style={
                  {
                    "bottom": 0,
                    "color": "transparent",
                    "height": "100%",
                    "left": 0,
                    "objectFit": undefined,
                    "objectPosition": undefined,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": "100%",
                  }
                }
              />
            </div>
          </a>
        </li>
        <li
          className="c5"
        >
          <a
            className="c6 c7"
            href="#test2"
            onClick={[Function]}
          >
            <div
              className="c8 c9"
            >
              <span
                className="c10"
              >
                Test Item 2
              </span>
              <span
                className="c11"
              >
                Test Subheading 2
              </span>
            </div>
            <div
              className="c12"
            >
              <img
                alt=""
                className="c13"
                data-nimg="fill"
                decoding="async"
                loading="lazy"
                onError={[Function]}
                onLoad={[Function]}
                src="https://res.cloudinary.com/mock-cloudinary-cloud/image/upload/v1707752509/icons/vk9xxxhnsltsickom6q9.svg"
                style={
                  {
                    "bottom": 0,
                    "color": "transparent",
                    "height": "100%",
                    "left": 0,
                    "objectFit": undefined,
                    "objectPosition": undefined,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": "100%",
                  }
                }
              />
            </div>
          </a>
        </li>
      </ul>
    </div>
  </nav>,
  ",",
]
`;
